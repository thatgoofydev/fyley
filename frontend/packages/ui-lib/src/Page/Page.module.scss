@import "src/base-styling/colors";
@import "src/base-styling/vars";

$page-header-height: 70px;

.pageContainer {
  width: 100%;
  height: calc(100% - #{$page-header-height});
  max-width: 1200px;
  margin: 0 auto;

  .pageHeader {
    height: 70px;

    h1 {
      margin: 0;
      line-height: 70px;
      font-size: 24px;
      font-weight: bold;
    }
  }

  .pageContentWrapper {
    background-color: $white;
    height: 100%;
    min-height: 600px;
    -webkit-box-shadow: 0 0 20px 0 rgba(0,0,0,0.16);
    -moz-box-shadow: 0 0 20px 0 rgba(0,0,0,0.16);
    box-shadow: 0 0 20px 0 rgba(0,0,0,0.16);
  }
}