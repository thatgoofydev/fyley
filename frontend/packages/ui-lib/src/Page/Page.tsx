import React, { FunctionComponent } from "react";
import styles from "./Page.module.scss";

export interface IPageProps {
  title: string;
}

export const Page: FunctionComponent<IPageProps> = ({ title, children }) => {
  return (
    <div className={styles.pageContainer}>
      <header className={styles.pageHeader}>
        <h1>{title}</h1>
      </header>
      <div className={styles.pageContent}>{children}</div>
    </div>
  );
};
