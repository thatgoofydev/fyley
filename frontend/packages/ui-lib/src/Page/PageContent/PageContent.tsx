import React, { FunctionComponent } from "react";
import styles from "./PageContent.module.scss";

export const PageContent: FunctionComponent = ({ children }) => {
  return <div className={styles.pageContent}>{children}</div>;
};
