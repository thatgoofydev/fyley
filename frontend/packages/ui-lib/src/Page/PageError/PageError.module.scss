@import "src/base-styling/colors";

.pageError {
  width: 450px;
  padding-top: 120px;
  margin: 0 auto;

  h1 {
    margin: 0;
    font-size: 48px;
  }

  h2 {
    margin: 10px 0;
    font-size: 24px;
    color: $secondary-text;
  }
}

.whatNext {
  font-size: 18px;

  p {
    margin-top: 32px;
    margin-bottom: 14px;
  }

  ul {
    margin: 0;
    padding-left: 30px;
    list-style: disc;
  }

  a {
    color: $blue;
  }
}

.details {
  margin-top: 20px;
  text-align: center;
  background-color: $grey;

  button {
    display: block;
    width: 100%;
    padding: 5px 10px;
    border: 0;
    outline: 0;
    background-color: transparent;
    font-size: 1em;
    cursor: pointer;
  }

  p {
    padding: 8px 16px;
    margin: 5px 0 0;
    text-align: left;
  }
}