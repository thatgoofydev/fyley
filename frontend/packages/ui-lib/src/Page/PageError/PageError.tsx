import React, { FunctionComponent, useState } from "react";
import styles from "./PageError.module.scss";

export interface IPageErrorProps {
  backLink?: string;
  details?: string;
}

export const PageError: FunctionComponent<IPageErrorProps> = ({ backLink, details }) => {
  const [showDetails, setDetailsVisible] = useState(false);

  const toggleDetails = () => setDetailsVisible(old => !old);

  return (
    <div className={styles.pageError}>
      <h1>Oops!</h1>
      <h2>Looks like something went wrong...</h2>
      <div className={styles.whatNext}>
        <p>What now?</p>
        <ul>
          <li>
            {backLink && (
              <>
                Go <a href={backLink}>back</a> and try again later.
              </>
            )}
            {!backLink && "Try again later."}
          </li>
          <li>Contact the application administrator.</li>
        </ul>
      </div>
      {details && (
        <div className={styles.details}>
          <button data-testid="error-toggle" onClick={toggleDetails}>
            {showDetails ? "Hide" : "Show"} details
          </button>
          {showDetails && <p>{details}</p>}
        </div>
      )}
    </div>
  );
};
