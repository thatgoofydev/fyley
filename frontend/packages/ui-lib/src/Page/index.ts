export { Page } from "./Page";
export { PageContent } from "./PageContent";
export { PageError } from "./PageError";
