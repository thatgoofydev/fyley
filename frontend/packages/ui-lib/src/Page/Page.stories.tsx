import React from "react";
import { Meta, Story } from "@storybook/react";
import { IPageProps, Page } from "./Page";
import { PageContent } from "./PageContent";

export default {
  title: "Pages/Page",
  component: Page
} as Meta;

export const PageStory: Story<IPageProps> = args => {
  return (
    <Page {...args}>
      <PageContent>
        <p>Some content</p>
        <p>Some content</p>
      </PageContent>
    </Page>
  );
};
PageStory.storyName = "Page";
PageStory.args = {
  title: "Page title"
};
