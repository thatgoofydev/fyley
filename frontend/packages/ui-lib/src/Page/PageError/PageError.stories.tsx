import React from "react";
import { Meta, Story } from "@storybook/react";
import { Page } from "../Page";
import { IPageErrorProps, PageError } from "./PageError";
import { PageContent } from "../PageContent/PageContent";

export default {
  title: "Pages/PageError",
  component: PageError
} as Meta;

export const PageErrorStory: Story<IPageErrorProps> = args => {
  return (
    <Page title="Page title">
      <PageContent>
        <PageError {...args} />
      </PageContent>
    </Page>
  );
};
PageErrorStory.storyName = "Message details";
PageErrorStory.args = {
  details: "some message giving more insight into the error",
  backLink: "#"
};
