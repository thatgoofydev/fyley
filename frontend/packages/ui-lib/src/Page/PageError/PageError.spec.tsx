import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Page } from "../Page";
import { PageError } from "./PageError";
import { PageContent } from "../PageContent/PageContent";

afterEach(cleanup);

describe("PageError", function () {
  it("should render", function () {
    const { container } = render(
      <Page title="Page title">
        <PageContent>
          <PageError />
        </PageContent>
      </Page>
    );

    expect(container).toMatchSnapshot();
  });

  describe("with details", function () {
    it("should render closed", function () {
      const { container } = render(
        <Page title="Page title">
          <PageContent>
            <PageError details="Some details about the error that occurred" />
          </PageContent>
        </Page>
      );

      expect(container).toMatchSnapshot();
    });

    it("should render opened when toggle clicked", function () {
      const { container, getByTestId, getByText } = render(
        <Page title="Page title">
          <PageContent>
            <PageError details="Some details about the error that occurred" />
          </PageContent>
        </Page>
      );

      getByTestId("error-toggle").click();
      expect(getByText("Some details about the error that occurred")).not.toBeNull();
      expect(container).toMatchSnapshot();
    });
  });
});
