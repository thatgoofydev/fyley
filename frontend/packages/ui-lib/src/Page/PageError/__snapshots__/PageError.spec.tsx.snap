// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageError should render 1`] = `
<div>
  <div
    class="pageContainer"
  >
    <header
      class="pageHeader"
    >
      <h1>
        Page title
      </h1>
    </header>
    <div
      class="pageContentWrapper"
    >
      <div
        class="pageError"
      >
        <h1>
          Oops!
        </h1>
        <h2>
          Looks like something went wrong...
        </h2>
        <div
          class="whatNext"
        >
          <p>
            What now?
          </p>
          <ul>
            <li>
              Try again later.
            </li>
            <li>
              Contact the application administrator.
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageError with details should render closed 1`] = `
<div>
  <div
    class="pageContainer"
  >
    <header
      class="pageHeader"
    >
      <h1>
        Page title
      </h1>
    </header>
    <div
      class="pageContentWrapper"
    >
      <div
        class="pageError"
      >
        <h1>
          Oops!
        </h1>
        <h2>
          Looks like something went wrong...
        </h2>
        <div
          class="whatNext"
        >
          <p>
            What now?
          </p>
          <ul>
            <li>
              Try again later.
            </li>
            <li>
              Contact the application administrator.
            </li>
          </ul>
        </div>
        <div
          class="details"
        >
          <button
            data-testid="error-toggle"
          >
            Show
             details
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageError with details should render opened when toggle clicked 1`] = `
<div>
  <div
    class="pageContainer"
  >
    <header
      class="pageHeader"
    >
      <h1>
        Page title
      </h1>
    </header>
    <div
      class="pageContentWrapper"
    >
      <div
        class="pageError"
      >
        <h1>
          Oops!
        </h1>
        <h2>
          Looks like something went wrong...
        </h2>
        <div
          class="whatNext"
        >
          <p>
            What now?
          </p>
          <ul>
            <li>
              Try again later.
            </li>
            <li>
              Contact the application administrator.
            </li>
          </ul>
        </div>
        <div
          class="details"
        >
          <button
            data-testid="error-toggle"
          >
            Hide
             details
          </button>
          <p>
            Some details about the error that occurred
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
